### PR TITLE
#337/1/UI/desktop/navbar

### PR DESCRIPTION
**Pull request by RobertS-ki**
Top navigation navbar with dropdown menu and responsivness.
obraz
obraz
obraz

### CODE REVIEW

File: client/src/Molecules/MenuItem/MenuItem.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import propTypes from 'prop-types';
+
+import MenuItemStyled from './MenuItemStyled';
+import NavLinkStyled from '../../Atoms/NavLinkStyled/NavLinkStyled';
+
+const MenuItem = (props) => {
+  const { text, linkTo, icon, stylefor } = props;
```
Comment (Smyku6) [inline client/src/Molecules/MenuItem/MenuItem.js:7]:
```js
const MenuItem = ({ text, linkTo, icon, stylefor }) => {
```

will be cleaner

File: client/src/Molecules/NavButton/NavButton.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  align-items: center;
+  transition: all 0.2s ease-in-out;
+  border-radius: 1.5625rem 0.625rem 0.625rem 1.5625rem;
+  :hover {
+    background-color: rgba(230, 230, 230, 0.4);
+  }
+`;
+
+const NavButton = (props) => {
+  const { menuOpen } = props;
```
Comment (Smyku6) [inline client/src/Molecules/NavButton/NavButton.js:20]:
```js
const NavButton = ({ menuOpen }) => {
```

will be cleaner

File: client/src/Molecules/MenuItem/MenuItem.js (hunk 1 above)
Comment (RobertS-ki) [reply client/src/Molecules/MenuItem/MenuItem.js:7]: True. Applied.

File: client/src/Molecules/NavButton/NavButton.js (hunk 2 above)
Comment (RobertS-ki) [reply client/src/Molecules/NavButton/NavButton.js:20]: Applied.

**Comment by vieraboschkova** [comment]
image
image
If the content of the navbar is supposed to be limited (very good thing to do), then set it to max and keep on all devices the same (here - navbar is wider on 1589 than on above 1600)

File: client/src/Organisms/Nav/Nav.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  const toggleHandler = () => {
+    setMenuOpen(!menuOpen);
+  };
+
+  let domNode = useClickOutside(() => {
+    setMenuOpen(false);
+  });
+
+  // when logged in:
+  if (isAuthenticated) {
```
Comment (vieraboschkova) [inline client/src/Organisms/Nav/Nav.js:36]: You could use ternany expression also to render one or another.

File: client/src/Molecules/NavCornerMenu/NavCornerMenu.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+const NavCornerMenu = () => {
+  const { dispatch } = useContext(UserDataContext);
+  const navigate = useNavigate();
+
+  return (
+    <NavCornerMenuStyled>
+      <MenuItem
+        stylefor="Cornermenu"
+        linkTo={`/settings`}
+        text="Settings"
```
Comment (vieraboschkova) [inline client/src/Molecules/NavCornerMenu/NavCornerMenu.js:22]: Polish/ english? 
image

File: client/src/Molecules/NavButton/NavButton.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    background-color: rgba(230, 230, 230, 0.5);
+  }
+`;
+
+const NavButton = ({ menuOpen }) => {
+  return (
+    <NavButtonStyled>
+      <ProfilePicture />
+      <div style={{ userSelect: 'none', whiteSpace: 'nowrap' }}>
+        Witaj, Matylda
```
Comment (vieraboschkova) [inline client/src/Molecules/NavButton/NavButton.js:26]: This should probably be showing the name of the user? Perhaps you can get it from the user context?

**Review by vieraboschkova — DISMISSED** [review]
Looks good! Great job! 
Check the dropdown menu and use user data to populate it, but apart from that - good to go!

File: client/src/Molecules/NavCornerMenu/NavCornerMenu.js (hunk 4 above)
Comment (RobertS-ki) [reply client/src/Molecules/NavCornerMenu/NavCornerMenu.js:22]: LoL. Polish.

File: client/src/Organisms/Nav/Nav.js (hunk 3 above)
Comment (RobertS-ki) [reply client/src/Organisms/Nav/Nav.js:36]: It seems more readable for me with if statement, than looking for ":" in the code.

File: client/src/Molecules/NavButton/NavButton.js (hunk 5 above)
Comment (RobertS-ki) [reply client/src/Molecules/NavButton/NavButton.js:26]: Fixed!

**Review by KatarzynaChmielecka — APPROVED** [review]

**Pull request merged**